### ISSUE TITLE
Search Sidebar Parameters Repopulating After Resetting Search

### ISSUE THREAD

**Issue by engerernoah** [tsgrp/HPI#2040]
When performing a search if the reset button is triggered and another search is run with different parameters, the values that were cleared out by the reset are repopulated. 

Here is a video of what is going on - taken on edge3: 
![searchResultsIssueEdge](https://user-images.githubusercontent.com/39744036/73015692-1c9a5100-3de2-11ea-9212-b4d68b7e89ee.gif)

This appears to only happen when enable saved search is NOT checked in the admin, but is likely tied to form support somehow.

**Comment by gsteimer** [comment]
@engerernoah - I just tried this on edge2 and wasn't seeing the problem.  Could this potentially be a form configuration issue?  There is a form option to "remember last search term".  Could that be on for this form field?

**Comment by engerernoah** [comment]
SEARCH SIDEBAR TABS - updating the logic surrounding the search sidebar tabs view. We removed the before render since this was instantiating multiple views and wiring up multiple listeners. This was causing the issue https://github.com/tsgrp/HPI/issues/2040 due to a function triggering twice that was updating the search parameters. Also we went to a more concrete approach to setting the tab views by explicitly handling the three search types we currently have (attribute, advanced and facet). Cleaned up linting errors and wrote unit tests for the tab view initialize, serialize and cleanup functions. Updated the mock_insurance_search_config.js file to have the search configs be backbone models to emulate our actual code. 

CR: skovacinski, jhof

REVISION: #24716 - OCMS

**Comment by engerernoah** [comment]
The fix for this led to another issue, reopening this since it will now be fixed with the revision used to fix issue: https://github.com/tsgrp/HPI/issues/2057

**Comment by engerernoah** [comment]
SEARCH NOT RENDERING ON REFRESH - Recent updates went into search sidebar to stop multiple renders from triggering code numerous time. This created a timing issue that caused searchSidebar to not render completely. We refactored the sidebar to handle more of its logic within the initialize instead of on beforeRender each time. We then added deferreds around the initialization of some of the subviews to ensure everything is set up before we render the sidebar. This allows us to only render the sidebar and all of its subviews once which fixes issues with the form showing twice and listeners being wired up more than once, while fixing the timing issues that caused the sidebar to render before all of the subviews were created. 

Additionally while working through this issue we updated the formatting in the attributesearch.js file be more consistent in order to enhance the readability. The only logic changes to attribute search were adding calls to stopListening() before wiring up our listeners just as good practice cleanup to safeguard against multiple listeners triggering in the future. These logic changes are on lines: 258 and 341

Unit tests were updated in searchSidebar to reflect the changes made, unit tests were not added to attributeSearch due to time constraints as the after render method would likely need to be refactored/ broken out to easily unit test. 

Revision: #24827

CR: jhof, rsaladrigas
